src/actions/authActions.js

```diff
@@ -102,6 +102,16 @@ export const updateFcmTokenAction = (walletId: string) => {
   };
 };
 
+export const proceedWithPkAction = (pin: string, action: (pk: string) => void) => {
+  return async (dispatch: Dispatch) => {
+    const { wallet: encryptedWallet } = await storage.get('wallet');
+    const saltedPin = await getSaltedPin(pin, dispatch);
+    const wallet = await decryptWallet(encryptedWallet, saltedPin);
+    const pk = get(wallet, 'signingKey.privateKey');
+    action(pk);
+  };
+};
```

Comment:
I think this logic is too complex. This should be basically a helper function that returns pk basing on pin (something like `getPrivateKeyFromPin = (pin: string) => { ...`). Then the `action` should be called outside of this function.

Reply:
Agree - moved this to utilities